spec/bson/big_decimal_spec.rb

```diff
@@ -0,0 +1,31 @@
+# encoding: utf-8
+require 'bigdecimal'
+require_relative "../spec_helper"
+require 'stringio'
+
+describe ::BigDecimal do
+  let(:a_number) { "4321.1234" }
+  let(:bson_number) { "Tt$\x97\x1F\xE1\xB0@".force_encoding(BSON::BINARY) }
+
+  subject { described_class.new(a_number) }
+
+  it "responds to to_bson" do
+    expect(subject).to respond_to(:to_bson)
+  end
+
+  it "to_bson returns a binary encoded number" do
+    expect(subject.to_bson).to eq(bson_number)
+  end
+
+  it "bson_type returns a binary encoded 1" do
+    expect(subject.bson_type).to eq("\x01".force_encoding(BSON::BINARY))
+  end
+
+  describe "class methods" do
+    it "builds a new BigDecimal from BSON" do
+      decoded = described_class.from_bson(StringIO.new(bson_number))
+      expect(decoded).to eq(subject)
```

Comment:
I don't agree. I feel that what you suggest is rspec overkill in this case. The right class expectation checks that `new()` surrounds the from bson conversion - this is something, that once in place is very very unlikely to change and therefore the expectation is very very likely to be met.  The right value expectation is different - it depends on what the Mongo ruby driver gives to the `from_bson` method and whether it can be decoded by the `from_bson` implementation.

But the above is moot because the test is not an integration test, we are not really getting a value from the Mongo ruby driver to use for the `from_bson` tests.

Further, the `from_bson` methods would only be used by a mongo-input plugin when ingesting data from mongo.

When we have a mongo input, these bson extensions and tests will need to move to a common place and a read integration test created in the new plugin, but a read integration test would not yield an instance of BigDecimal or Timestamp because the type encoded for BigDecimal is a Float so we would get a Float back, Timestamp is encoded as a Time type so we would get a Time back.

In conclusion, the `from_bson` methods will never be used at all.  So the tests for these methods do not mean any more than "was the code sort-of correct at the time of writing".

I can't see any reason to change the tests.